processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayConnector.java

```diff
@@ -55,8 +62,16 @@ public JsonNode connectorPayload(T gateway, String topicName) {
             definition.set(PROCESSORS_PARAMETER, processors);
         }
 
-        addConnectorSpecificPayload(gateway, topicName, definition);
+        Map<String, String> sensitiveParameters;
+        if (expectsSensitiveParameters()) {
```

Comment:
It's a shame we cannot somehow leverage the sensitive parameter names from the resolved action to be able to check we not only have a stored secret for the connector but also it contains the expected parameters. At the moment the implementations of `AbstractGatewayConnectoraddConnectorSpecificPayload(..)` could retrieve `null` values and break at runtime.